yb-voyager/src/dbzm/valueConverter.go

```diff
@@ -0,0 +1,142 @@
+limitations under the License.
+*/
+package dbzm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+type ValueConverter interface {
+	ConvertRow(tableName string, columnNames []string, row string) (string, error)
+	ConvertEvent(ev *tgtdb.Event, table string) error
+}
+
+func NewValueConverter(exportDir string, tdb tgtdb.TargetDB, snapshotMode bool) ValueConverter {
+	if utils.IsDebeziumForDataExport(exportDir) {
+		return NewDebeziumValueConverter(exportDir, tdb, snapshotMode)
+	} else {
+		return &NoOpValueConverter{}
+	}
+}
+
+//============================================================================
+
+type NoOpValueConverter struct{}
+
+func (nvc *NoOpValueConverter) ConvertRow(tableName string, columnNames []string, row string) (string, error) {
+	return row, nil
+}
+
+func (nvc *NoOpValueConverter) ConvertEvent(ev *tgtdb.Event, table string) error {
+	return nil
+}
+
+//============================================================================
+
+type DebeziumValueConverter struct {
+	schemaRegistry      *SchemaRegistry
+	valueConverterSuite map[string]tgtdb.ConverterFn
+	converterFnCache    map[string][]tgtdb.ConverterFn
+	snapshotMode        bool
+}
+
+func NewDebeziumValueConverter(exportDir string, tdb tgtdb.TargetDB, snapshotMode bool) *DebeziumValueConverter {
+	schemaRegistry := NewSchemaRegistry(exportDir)
+	tdbValueConverterSuite := tdb.GetDebeziumValueConverterSuite(snapshotMode)
+
+	return &DebeziumValueConverter{
+		schemaRegistry:      schemaRegistry,
+		valueConverterSuite: tdbValueConverterSuite,
+		converterFnCache:    map[string][]tgtdb.ConverterFn{},
+		snapshotMode:        snapshotMode,
+	}
+}
+
+func (conv *DebeziumValueConverter) ConvertRow(tableName string, columnNames []string, row string) (string, error) {
+	converterFns, err := conv.getConverterFnCache(tableName, columnNames)
+	if err != nil {
+		return row, err
+	}
+	columnValues := strings.Split(row, "\t")
+	var transformedValues []string
+	for i, columnValue := range columnValues {
+		if columnValue != "\\N" {
+			transformedValue, err := converterFns[i](columnValue)
+			if err != nil {
+				return row, err
+			}
+			transformedValues = append(transformedValues, transformedValue)
+		} else {
+			transformedValues = append(transformedValues, columnValue)
+		}
+	}
+	transformedRow := strings.Join(transformedValues, "\t")
+	return transformedRow, nil
+}
+
+func (conv *DebeziumValueConverter) getConverterFnCache(tableName string, columnNames []string) ([]tgtdb.ConverterFn, error) {
+	result := conv.converterFnCache[tableName]
+	if result == nil {
+		colSchemas, err := conv.schemaRegistry.GetColumnSchemas(tableName, columnNames)
+		if err != nil {
+			return nil, err
+		}
+		result = make([]tgtdb.ConverterFn, len(columnNames))
+		for i := range columnNames {
+			schema := colSchemas[i]
+			result[i] = schema.ColDbzSchema.GetConverterFn(conv.valueConverterSuite)
```

Comment:
Get rid of the `GetConverterFn()` -- that's too much engineering ;-) . Directly lookup the type name in the converter suite.